sources/csv_reader/__init__.py

```diff
@@ -0,0 +1,64 @@
+"""
+Source, which uses the `filesystem` source and DuckDB
+to extract CSV files data from the given locations.
+"""
+from typing import Iterable, List
+
+import duckdb
+import pendulum
+
+import dlt
+from dlt.common.typing import TDataItem, TAnyDateTime
+from filesystem import filesystem
+from fsspec.implementations.local import LocalFileSystem
+
+from .helpers import add_columns
+
+
+@dlt.resource
+def read_location(files):
+    """A resource to extract data from the given CSV files.
+
+    Args:
+        files (List[FileItem]): A list of files to read.
+
+    Returns:
+        Iterable[TDataItem]: Data items, read from the given CSV files.
+    """
+    state = dlt.current.resource_state()
+    start_from = state.setdefault("last_modified", pendulum.datetime(1970, 1, 1))
+
+    results = []
+    for file in files:
+        if file["modification_date"] <= start_from:
+            continue
+
+        file_res = duckdb.sql(f"""SELECT * FROM read_csv_auto('{file["file_url"]}')""")
+        results += add_columns(file_res.columns, file_res.fetchall())
+
+        state["last_modified"] = max(file["modification_date"], state["last_modified"])
+
+    yield results
+
+
+@dlt.source
+def csv_reader(bucket: str, globs: List[str] = ("*",)) -> Iterable[TDataItem]:
+    """
+    A source to extract data from CSV files from
+    one or several locations.
+
+    Args:
+        bucket (str): A bucket URL.
+        globs (Optional[List[str]]):
+            A list of glob patterns to match files.
+            Every glob will be extracted into a separate table.
+
+    Returns:
+        Iterable[TDataItem]:
+            Data items, read from the matched CSV files.
+    """
+    duckdb.register_filesystem(LocalFileSystem())
```

Comment:
This item is very unpleasant - `duckdb` requires a filesystem registration. Looks like we need, looking at the bucket scheme, decide which filesystem to register.

Reply:
Should we pass filesystem via parameters and make it swappable, if not passed then maybe fallback to `LocalFileSystem` or otherwise just stop the pipeline?